dask/optimization.py

```diff
@@ -808,3 +895,40 @@ def key_split(s):
             return result
     except Exception:
         return 'Other'
+
+
+class SubgraphCallable(object):
+    """Create a callable object from a dask graph.
+
+    Parameters
+    ----------
+    dsk : dict
+        A dask graph
+    outkey : hashable
+        The output key from the graph
+    inkeys : list
+        A list of keys to be used as arguments to the callable.
+    name : str, optional
+        The name to use for the function.
+    """
+    __slots__ = ('dsk', 'outkey', 'inkeys', 'name')
+
+    def __init__(self, dsk, outkey, inkeys, name='subgraph_callable'):
+        self.dsk = dsk
+        self.outkey = outkey
+        self.inkeys = inkeys
+        self.name = name
+
+    def __repr__(self):
+        return self.name
+
+    def __call__(self, *args):
+        if not len(args) == len(self.inkeys):
+            raise ValueError("Expected %d args, got %d"
+                             % (len(self.inkeys), len(args)))
+        return _get_recursive(self.dsk, self.outkey,
+                              dict(zip(self.inkeys, args)))
```

Comment:
I recommend using `dask.local.get_sync`.  The `dask.core.get` function is really only around due to history.  It can be very inefficient.

Reply:
For graphs of the size we end up fusing, and with the improvements made here to the recursive version, using `dask.core.get` is much faster than `get_sync`.

Reply:
Hrm, interesting.  I look forward to inspecting that further.